clients/unshaded/src/main/java/tachyon/client/RemoteBlockInStream.java

```diff
@@ -412,4 +437,23 @@ private boolean updateCurrentBuffer() throws IOException {
     }
     return false;
   }
+
+  /**
+   * Clear the previous reader, release the resource it references.
+   *
+   * @return true if reader is successfully cleared or no clearing is needed
+   */
+  private void closeReader() throws IOException {
+    if (mStandingReader != null) {
+      try {
+        //TODO: we should implement close() in all readers.
```

Comment:
I think this comment is similar to "Maybe RemoteBlockReader should implement Closeable". Please find my reply below.